verifiable-db/src/query/aggregation/mod.rs

```diff
@@ -217,6 +224,83 @@ impl NodeInfo {
         )
     }
 }
+
+#[derive(Clone, Debug)]
+pub(crate) struct NodeInfoTarget {
+    /// The hash of the embedded tree at this node. It can be the hash of the row tree if this node is a node in
+    /// the index tree, or it can be a hash of the cells tree if this node is a node in a rows tree
+    pub(crate) embedded_tree_hash: HashOutTarget,
+    /// Hashes of the children of the current node, first left child and then right child hash. The hash of left/right child
+    /// is the empty hash (i.e., H("")) if there is no corresponding left/right child for the current node
+    pub(crate) child_hashes: [HashOutTarget; 2],
+    /// value stored in the node. It can be a primary index value if the node is a node in the index tree,
+    /// a secondary index value if the node is a node in a rows tree
+    pub(crate) value: UInt256Target,
+    /// minimum value associated to the current node. It can be a primary index value if the node is a node in the index tree,
+    /// a secondary index value if the node is a node in a rows tree
+    pub(crate) min: UInt256Target,
+    /// minimum value associated to the current node. It can be a primary index value if the node is a node in the index tree,
+    /// a secondary index value if the node is a node in a rows tree
+    pub(crate) max: UInt256Target,
+}
+
+impl NodeInfoTarget {
+    pub(crate) fn build(b: &mut CBuilder) -> Self {
+        let [value, min, max] = b.add_virtual_u256_arr();
+        let [left_child_hash, right_child_hash, embedded_tree_hash] =
+            array::from_fn(|_| b.add_virtual_hash());
+        Self {
+            embedded_tree_hash,
+            child_hashes: [left_child_hash, right_child_hash],
+            value,
+            min,
+            max,
+        }
+    }
+
+    /// Build an instance of `Self` without range-check the `UInt256Target`s
```

Comment:
It's up to the caller to determine whether it's safe to use this method or not, depending on how the values are employed. For instance, if the node info is employed to compute the hash of the node, it should be safe to allocate without range checks since the hash will be different in case the prover provides out-of-range limbs.